pkg/controller/resourcequota/resource_quota_controller.go

```diff
@@ -79,9 +98,14 @@ type ResourceQuotaController struct {
 	resyncPeriod controller.ResyncPeriodFunc
 	// knows how to calculate usage
 	registry quota.Registry
+	// knows how to monitor all the resources tracked by quota and trigger replenishment
+	quotaMonitor *QuotaMonitor
+	// controls the workers that process quotas
+	workerLock sync.RWMutex
```

Comment:
This is used as a really big hammer.  Please add more detail in your comment.  You're using this to control access to write the quotaMonitors?